src/plugin/plugin-manager.js

```diff
@@ -82,15 +82,29 @@ export default class PluginManager {
       throw new Error(Error.Severity.RECOVERABLE, Error.Category.PLAYER, Error.Code.RUNTIME_ERROR_NOT_REGISTERED_PLUGIN, name);
     }
     let pluginClass = PluginManager._registry.get(name);
-    if (pluginClass && pluginClass.isValid()) {
+    this._maybeDisablePlugin(pluginClass, config.disable);
+    if (pluginClass && pluginClass.isValid() && !pluginClass.disabled) {
       this._plugins.set(name, pluginClass.createPlugin(name, player, config));
       logger.debug(`Plugin <${name}> has been loaded`);
       return true;
     }
-    logger.debug(`Plugin <${name}> isn\'t loaded, isValid()=false`);
+    logger.debug(`Plugin <${name}> isn\'t loaded, isValid()=${(!!pluginClass && pluginClass.isValid()).toString()}, disabled=${(!!pluginClass && pluginClass.disabled).toString()}`);
     return false;
   }
 
+  /**
+   * sets the disabled property.
+   * @param {Function} pluginClass - the current plugin
+   * @param {boolean} disable - config
+   * @returns {void}
+   * @private
+   */
+  _maybeDisablePlugin(pluginClass: ?Function, disable: boolean): void {
+    if (typeof disable !== 'undefined' && pluginClass) {
+      pluginClass.disabled = disable;
```

Comment:
there's a reason that we save this state?

Reply:
i debated myself if it's an overkill. but i think it's good to save the state of the plugin and not to rely solely on the config. we might want to disable the plugin from within (e.g. disable analytic s on offline mode). another use case i see is when the config of the plugin is not reachable / complex to get.

Reply:
@odedhutzler I think offline is something that might change across sessions while disable is deterministic so we can indeed simplify this and just check if plugin config has disable true and just not initiate it.
There's no added value in setting the statics of the class, as plugins in player are only used as instance on a player instance.
Think about it this way, if you update the statics of the class another player might be instantiated and static disabled of it will also be true, although it wasn't set like that in the setup of it.

It's enough to reduce this to a conditional check in the `load` method and call it a day.